### PR TITLE
Check ruby version 2

### LINKED ISSUE #112: Cached gems layer reused even when ruby minor version changes
## What happened?

We built a Ruby app with version `2.7.2` of mri, then rebuilt it, changing the version to `2.6.6`. The app failed to run after rebuild because it was not able to find the required gems.

## Reproduction Steps

1. Clone [this sample app](https://github.com/paketo-buildpacks/samples/tree/main/ruby/puma)
2. Generate a `Gemfile.lock` by running `bundle install` locally.
3. Add a `buildpack.yml` that specifies an MRI version:
```yaml
---

mri:
  version: 2.7.2
```
4. Build the app:
```shell
pack build changing-version --builder paketobuildpacks/builder:0.1.20-base --buildpack gcr.io/paketo-buildpacks/ruby:0.1.5 --clear-cache
```
5. Run the app
```
docker run -d --env PORT=8080 --publish 8080:8080 changing-version
```
6. And see the expected output when accessing it:
```
curl 0.0.0.0:8080
<!DOCTYPE html>
  <html>
    <head>
      <title>Powered By Paketo Buildpacks</title>
    </head>
    <body>
      <img style="display: block; margin-left: auto; margin-right: auto; width: 50%;" src="https://paketo.io/images/paketo-logo-full-color.png"></img>
    </body>
  </html>
```

7. Then, change the `buildpack.yml` to specify a different MRI version:
```yaml
---

mri:
  version: 2.6.6
```
8. Rebuild the app without clearing the cache:
```shell
 pack build changing-version --builder paketobuildpacks/builder:0.1.20-base --buildpack gcr.io/paketo-buildpacks/ruby:0.1.5
```

9. Note that build output includes:
```
[builder] Paketo Bundle Install Buildpack 0.1.0
[builder]   Reusing cached layer /layers/paketo-buildpacks_bundle-install/gems
[builder]
```
10. Re-run the app (in interactive mode); note the error output:
```
docker run -it --env PORT=8080 --publish 8080:8080 changing-version
bundler: command not found: puma
Install missing gem executables with `bundle install`
```
11. Re-run the app and jump into the container; note that the gems are in a `ruby/2.7.0` directory
```shell
docker run -it --env PORT=8080 --publish 8080:8080  --entrypoint=launcher changing-version  /bin/bash
cnb@80bec5a816a0:/workspace$ ll /layers/paketo-buildpacks_bundle-install/gems/ruby/
total 12
drwxr-xr-x 3 cnb cnb 4096 Jan  1  1980 ./
drwxr-xr-x 4 cnb cnb 4096 Jan  1  1980 ../
drwxr-xr-x 9 cnb cnb 4096 Jan  1  1980 2.7.0/
```
The buildpack should rebuild the gem layer when the minor version of ruby changes, because the gems will need to be placed in a different subdirectory.

## Checklist

Please confirm the following:

* [x] I have included log output.
* [x] The log output includes an error message.
* [x] I have included steps for reproduction.

### PR DESCRIPTION
**Pull request by sophiewigmore**
Signed-off-by: Forest Eckhardt <feckhardt@pivotal.io>

## Summary

Resolves #112. 

Performs a check on the Ruby version during each build by running `ruby --version` and checking if it differs from the previous build, if there was one. This should cover the case in which we rebuild an app with a different Ruby version (i.e. with `$BP_MRI_VERSION` set), but the `Gemfile` and `Gemfile.lock` stay the same. 

## Use Cases

This ensure that the cached `bundle-install` layer will NOT be reused if the Ruby version changes between builds.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have added an integration test, if necessary.

### CODE REVIEW

**Review by ryanmoran — CHANGES_REQUESTED** [review]
I'm not sure this was clear from the original issue, but I think that we only want to be rebuilding if the major or minor version changes. I was able to trigger a rebuild by only changing the patch version and I believe that is a bit too aggressive. Would it be possible to parse the version number and still reuse the cached layer if the change was only for a minor version bump?

**Comment by sophiewigmore** [comment]
> Would it be possible to parse the version number and still reuse the cached layer if the change was only for a minor version bump?

Do you mean  reuse the cached layer if the change was a **patch**? But then trigger a rebuild on major/minor version changes?

**Comment by ryanmoran** [comment]
@sophiewigmore yes. This is what I meant to say.

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,8 @@ go 1.14
 
 require (
 	github.com/BurntSushi/toml v0.3.1
+	github.com/Masterminds/semver v1.5.0
+	github.com/blang/semver v3.5.0+incompatible
```
Comment (ryanmoran) [inline go.mod:8]:
```suggestion
```

I think this may have been pulled in accidentally.

**Review by ryanmoran — CHANGES_REQUESTED** [review]

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**